p2p/src/simulated/network.rs

```diff
@@ -34,6 +34,43 @@ use tracing::{debug, error, trace, warn};
 /// Task type representing a message to be sent within the network.
 type Task<P> = (Channel, P, Recipients<P>, Bytes, oneshot::Sender<Vec<P>>);
 
+/// Target for a message in a split receiver.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SplitTarget {
+    Primary,
+    Secondary,
+    Both,
+}
+
+/// Origin of a message in a split sender.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SplitOrigin {
+    Primary,
+    Secondary,
+}
+
+/// A function that forwards messages from [SplitOrigin] to [Recipients].
```

Comment:
Opted to support both split forwarding/routing because we will want to map each twin to some unique set of participants (each peer should only get a message from one twin).